UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/GroupContract.java

```diff
@@ -0,0 +1,219 @@
+        }
+
+        Group group;
+        try {
+            group = cUtil.getState(stub, groupId, Group.class);
+        } catch (LedgerAccessError e) {
+            return e.getJsonError();
+        }
+
+        if(!(group.getUserList().contains(enrollmentId))){
+            return GsonWrapper.toJson(cUtil.getUserNoRemoveError(enrollmentId, groupId));
+        }
+            group.getUserList().remove(enrollmentId);
+
+        // check approval
+        List<String> requiredIds = Collections.singletonList(enrollmentId);
+        List<String> requiredTypes = Collections.singletonList("admin");
+        if (!cUtil.validateApprovals(
+                ctx,
+                requiredIds,
+                requiredTypes,
+                this.contractName,
+                "removeUserFromGroup",
+                Collections.singletonList(groupId))) {
+            return GsonWrapper.toJson(cUtil.getInsufficientApprovalsError());
+        }
+
+        cUtil.putAndGetStringState(stub, groupId, GsonWrapper.toJson(group));
+
+        // success
+        return "";
+    }
+
+    /**
+     * Removes a user from all groups
+     * @param ctx transaction context providing access to ChaincodeStub etc.
+     * @param enrollmentId identifier of user to remove
+     * @return empty string on success, serialized error on failure
+     */
+    @Transaction()
+    public String removeUserFromAllGroups(final Context ctx, String enrollmentId) {
+        ChaincodeStub stub = ctx.getStub();
+
+        ArrayList<InvalidParameter> invalidParams = cUtil.getParameterErrorsForEnrollmentId(enrollmentId);
+        if (!invalidParams.isEmpty()) {
+            return GsonWrapper.toJson(cUtil.getUnprocessableEntityError(invalidParams));
+        }
+
+        // check approval
+        List<String> requiredIds = Collections.singletonList(enrollmentId);
+        List<String> requiredTypes = Collections.singletonList("admin");
+        if (!cUtil.validateApprovals(
+                ctx,
+                requiredIds,
+                requiredTypes,
+                this.contractName,
+                "removeUserFromAllGroups",
+                Collections.singletonList(enrollmentId))) {
+            return GsonWrapper.toJson(cUtil.getInsufficientApprovalsError());
+        }
+
+        cUtil.getGroupsForUser(stub, enrollmentId).forEach(item ->{
+            item.getUserList().remove(enrollmentId);
+            cUtil.putAndGetStringState(stub, item.getGroupId(),GsonWrapper.toJson(item));
+        });
+
+        // success
+        return "";
+    }
+
+    /**
+     * Gets GroupList from the ledger.
+     * @param ctx transaction context providing access to ChaincodeStub etc.
+     * @return Serialized List of Matching Groups on success, serialized error on failure
+     */
+    @Transaction()
+    public String getAllGroups(final Context ctx) {
+        ChaincodeStub stub = ctx.getStub();
+
+
+        List<Group> groupIdList = cUtil.getAllGroups(stub);
```

Comment:
```suggestion
        List<Group> groupList = cUtil.getAllGroups(stub);
```
This is not a group-id list. Is this intended?